### PR TITLE
fix(gatsby): enable ipc messaging on develop

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

This opens the door to enable jobs API for develop and not just `gatsby build` this was broken when develop got reworked.

You can test it with installing `yarn add gatsby@ipc-develop` and running it on https://github.com/wardpeet/gatsby-ipc-tester

### CODE REVIEW

**Comment by ascorbic** [comment]
We like this for Desktop too!

File: packages/gatsby/src/commands/develop.ts [hunk 1]
```diff
@@ -155,6 +155,14 @@ class ControllableScript {
     }
     this.process.on(`exit`, callback)
   }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  send(msg: any): void {
+    if (!this.process) {
+      throw new Error(`Trying to attach exit handler before process starter`)
```
Comment (pieh) [inline packages/gatsby/src/commands/develop.ts:161]: NIT :)
```suggestion
      throw new Error(`Trying to attach exit handler before process started`)
```

Comment (pieh) [reply packages/gatsby/src/commands/develop.ts:161]: And well, this msg doesn't make sense - we don't try to attach exit handlers - is this copied from somewhere? :P

File: integration-tests/structured-logging/__tests__/ipc-send.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+  afterAll(async () => {
+    gatsbyProcess.kill()
+  })
+
+  // TODO remove to a more global integration test but for now we want to make sure it gets an external send command
+  it(`should kill the develop process when exit command is given`, done => {
+    expect.assertions(1)
+    gatsbyProcess.on("exit", () => {
+      expect(true).toBe(true)
```
Comment (pieh) [inline integration-tests/structured-logging/__tests__/ipc-send.js:35]: true ...true ... ;)

please ignore this - I get why, just couldn't not comment on this :)

**Pull request merged**